hugie/endpoint.py

```diff
@@ -84,7 +107,39 @@ def create(
         data (str): Path to JSON data to create the endpoint
     """
 
-    data = InferenceEndpointConfig.from_json(data).model_dump()
+    if not data:
+        data = {
+            "accountId": None,
+            "name": name,
+            "type": type,
+            "compute": {
+                "accelerator": accelerator,
+                "instanceType": instanceType,
+                "instanceSize": instanceSize,
+                "scaling": {
+                    "minReplica": minReplica,
+                    "maxReplica": maxReplica,
+                },
+            },
+            "model": {
+                "repository": repository,
+                "revision": revision,
+                "task": task,
+                "framework": framework,
+                "image": {
+                    "huggingface": {},
+                },
+            },
+            "provider": {
+                "vendor": vendor,
+                "region": region,
+            },
+            "type": "protected",
+        }
+        data = InferenceEndpointConfig.parse_obj(data).model_dump()
+    else:
+        data = InferenceEndpointConfig.from_json(data).model_dump()
```

Comment:
Will this ever get triggered given you have the initial `if not data`?

Reply:
It will if you pass the config only

Reply:
ahh my bad i didn't see the deleted line :+1: